src/console_cmds.cpp

```diff
@@ -614,6 +615,118 @@ DEF_CONSOLE_CMD(ConBanList)
 	return true;
 }
 
+/**
+* Loads the ban list from the specified path.
+* @param path The path to load the ban list from.
+* @return The number of entries loaded from the file or -1 if an error occurred while loading entries.
+*/
+static int LoadBanList(const std::string& path)
+{
+	const char* const group_names[] = {
+		"bans",
+		nullptr
+	};
+
+	const auto ini = new IniFile(group_names);
+	ini->LoadFromDisk(path, NO_DIRECTORY);
+	const auto ban_group = ini->GetGroup("bans", false);
+
+	auto loaded_count = 0;
+	auto is_loaded = ban_group != nullptr;
+	if (is_loaded) {
+		loaded_count = 0;
+		for (const IniItem *item = ban_group->item; item != nullptr; item = item->next) {
+			// banned clients are added to global ban list when kicked or banned
+			NetworkServerKickOrBanIP(item->name.c_str(), true, nullptr);
+			loaded_count++;
+		}
+	}
+
+	delete ini;
+	return is_loaded ? loaded_count : -1;
+}
+
+/**
+* Loads a ban list from a user-specified file. The entries in the list are added to the current collection of banned entries.
+* @see ConBanList
+*/
+DEF_CONSOLE_CMD(ConBanListLoad)
+{
+	if (argc == 0) {
+		IConsoleHelp("Loads the IP's of banned clients from a file: Usage 'banlist_load <path_to_file>'");
+		return true;
+	}
+
+	if (argc != 2) {
+		return false;
+	}
+
+	const auto path = argv[1];
+	IConsolePrintF(CC_INFO, "Loading banlist from '%s'...", path);
+	auto loaded_entries_count = LoadBanList(path);
+	if (loaded_entries_count == -1) {
+		IConsolePrint(CC_WARNING, "File does not exist or [bans] section is unspecified.");
+		return false;
+	}
+
+	IConsolePrintF(CC_INFO, "Loaded %d entries.", loaded_entries_count);
+	return true;
+}
+
+/**
+* Saves the specified ban list to the specified path.
+* @param path The path to save the ban list to.
+* @param ban_list The ban list to save.
+* @return The number of entries saved to the file or -1 if an error occurred while saving entries.
+*/
+static int SaveBanList(const std::string& path, const StringList& ban_list)
+{
+	const char* const group_names[] = {
+		"bans",
+		nullptr
+	};
+
+	const auto ini = new IniFile(group_names);
+	const auto ban_group = ini->GetGroup("bans", true);
+
+	auto item_count = 0;
+	for (const auto name : ban_list) {
```

Comment:
As the clang compiler complains, you are making copies of all the strings in the list during this loop. Type deductions elide references so you need to be explicit if you want a deduced type to be a reference. Better just be explicit about the type.
```suggestion
	for (const std::string &name : ban_list) {
```